### PR TITLE
C.3.2 view course

### LINKED ISSUE #94: C-3-2-ViewCourses

### LINKED ISSUE #95: C-3-2-1-ViewOneCourse

### LINKED ISSUE #96: C-3-3-HideCourses*FromMarket*

### LINKED ISSUE #97: C-3-4-1 Add lesson to course

### LINKED ISSUE #111: C.3.15 Create Content Difficulty Rating

### LINKED ISSUE #112: C.3.16 Update content difficulty rating

### LINKED ISSUE #113: D.2.1 View available courses

### LINKED ISSUE #114: D.2.2 View Course content

### LINKED ISSUE #118: E.3.1 Add comment to video

### PR DESCRIPTION
**Pull request by kengthong**
### Changelog:
- feature: get all sensei courses 
- get course [returns course contracts too depending on the type of user]

## Description:
_Eg; Fixed router bug_
## Checklist:
- [x] Merged latest develop
- [x] PR title makes sense

### CODE REVIEW

File: src/controllers/course.controller.ts [hunk 1]
```diff
@@ -64,6 +64,89 @@ export class CourseController {
...
+  }
+
+  /** Get one course
+   *  if student => get course without contract
+   *  if sensei/admin => get course with contracts
+   * */
+  public static async getOneCourse(req, res) {
+    const { courseId } = req.params;
+    const { user } = req;
+    console.log('user =', user);
```
Comment (ianleck) [inline src/controllers/course.controller.ts]: Console log

File: src/services/course.service.ts [hunk 2]
```diff
@@ -119,6 +132,81 @@ export default class CourseService {
...
+          model: User,
+          attributes: ['firstName', 'lastName', 'profileImgUrl', 'occupation'],
+        },
+        CourseContract,
+      ],
+    });
+    return courses;
+  }
+
+  public static async getOneCourse(courseId: string, user?) {
```
Comment (ianleck) [inline src/services/course.service.ts]: Actually can I check what is in the { user } object in req

File: src/services/course.service.ts [hunk 3]
```diff
@@ -119,6 +132,81 @@ export default class CourseService {
...
+    const _user = user ? await User.findByPk(user.accountId) : null;
+    const courseWithoutContracts = await Course.findByPk(courseId, {
+      include: [
+        Category,
+        {
+          model: User,
+          attributes: ['firstName', 'lastName', 'profileImgUrl', 'occupation'],
```
Comment (ianleck) [inline src/services/course.service.ts:195]: Do we need to include accountId in case they want to click on him to see his profile

Comment (ianleck) [reply src/services/course.service.ts:195]: Same for the other gets

**Review by ianleck — CHANGES_REQUESTED** [review]
Reply comments

File: src/services/course.service.ts (hunk 2 above)
Comment (kengthong) [reply src/services/course.service.ts]: the user from jwt token, maybe i shld pass accountId instead

File: src/services/course.service.ts (hunk 3 above)
Comment (kengthong) [reply src/services/course.service.ts:195]: course has accountId if im not mistaken

**Review by ianleck — APPROVED** [review]

**Pull request merged**